app/components/novnc-rfb/component.js

```diff
@@ -0,0 +1,34 @@
+import Component from '@ember/component';
+import ENV from 'irene/config/environment';
+import RFB from '@novnc/novnc/core/rfb';
+
+export default Component.extend({
+  rfb: null,
+  didInsertElement(){
+    const canvasEl = this.element.getElementsByClassName("canvas-container")[0];
+    const deviceToken = this.get("file.deviceToken");
+    const deviceFarmURL =
+      `wss://${ENV.deviceFarmHost}/${ENV.deviceFarmPath}?token=${deviceToken}`
```

Comment:
or send the url rather

Reply:
Also do not generate the url. It can the use ws or. have a port number etc. Check if the url can be gotten from config directly and append just the token